docs/learn.md

```diff
@@ -721,20 +723,1250 @@ template](https://github.com/melange-re/melange-opam-template).
+
+#### JavaScript classes
+
+JavaScript classes are special kinds of objects. To interact with classes,
+Melange exposes `bs.new` to emulate e.g. `new Date()`:
+
+```ocaml
+type t
+external create_date : unit -> t = "Date" [@@bs.new]
+let date = create_date ()
+```
+
+Which generates:
+
+```js
+var date = new Date();
+```
+
+You can chain `bs.new` and `bs.module` if the JavaScript class you want to work
+with is in a separate JavaScript module:
+
+```ocaml
+type t
+external book : unit -> t = "Book" [@@bs.new] [@@bs.module]
+let myBook = book ()
+```
+
+Which generates:
+
+```js
+var Book = require("Book");
+var myBook = new Book();
+```
+
+### Bind to global JavaScript functions
+
+Binding to a JavaScript function makes use of `external`, like with objects. If
+we want to bind to a function available globally, Melange offers the `bs.val`
+attribute:
+
+```ocaml
+external imul : int -> int -> int = "Math.imul" [@@bs.val]
+```
+
+Or for `document`:
+
+```ocaml
+(* Abstract type for the `document` value *)
+type document
+
+external document : document = "document" [@@bs.val]
+```
+
+#### Using functions from other JavaScript modules
+
+`bs.module` is like the `bs.val` attribute, but it accepts a string with the
+name of the module, or the relative path to it.
+
+```ocaml
+external dirname : string -> string = "dirname" [@@bs.module "path"]
+let root = dirname "/User/github"
+```
+
+Generates:
+
+```js
+var Path = require("path");
+var root = Path.dirname("/User/github");
+```
+
+#### Labeled arguments
+
+OCaml has [labeled arguments](https://v2.ocaml.org/manual/lablexamples.html),
+which can also be optional, and work with `external` as well.
+
+Labeled arguments can be useful to provide more information about the arguments
+of a JavaScript function that is called from Melange.
+
+Let’s say we have the following JavaScript function, that we want to call from
+Melange:
+
+```js
+// MyGame.js
+
+function draw(x, y, border) {
+  // let’s assume `border` is optional and defaults to false
+}
+draw(10, 20)
+draw(20, 20, true)
+```
+
+When writing Melange bindings, we can add labeled arguments to make things more
+clear:
+
+```ocaml
+external draw : x:int -> y:int -> ?border:bool -> unit -> unit = "draw"
+  [@@module "MyGame"]
+
+let () = draw ~x:10 ~y:20 ~border:true ()
+let () = draw ~x:10 ~y:20 ()
+```
+
+Generates:
+
+```js
+var MyGame = require("MyGame");
+
+MyGame.draw(10, 20, true);
+MyGame.draw(10, 20, undefined);
+```
+
+The generated JavaScript function is the same, but now the usage in Melange is
+much clearer.
+
+**Note**: in this particular case, a final param of type unit, `()` must be
+added after `border`, since `border` is an optional argument at the last
+position. Not having the last param `unit` would lead to a warning, which is
+explained in detail [in the OCaml
+documentation](https://ocaml.org/docs/labels#warning-this-optional-argument-cannot-be-erased).
+
+Note that you can freely reorder the labels on the Melange side, they will
```

Comment:
on the application side. but the definition side must match the order of arguments in JS. I think it's important to mention that.

Reply:
Fixed in https://github.com/melange-re/documentation-site/pull/12/commits/4d2c9ba756c3c28304e12ff8eacbc092a8ccd9ab.